命令行库无法构建通过 CommandLine.As.g.cs error CS0234

提示错误如下

DotNetCampus.CommandLine.Analyzer\DotNetCampus.CommandLine.Generators.InterceptorGenerator\CommandLine.Interceptors\CommandLine.As.g.cs(14,23,14,71): error CS0234: 命名空间“HalnerefaiwhoLanallwaynai”中不存在类型或命名空间名“OptionsBuilder”(是否缺少程序集引用?)

最简复现 Demo 代码如下

```csharp
// Options.cs

namespace HalnerefaiwhoLanallwaynai;

public class Options
{
    public required string Name { get; init; }
}
```

```csharp
// Program.cs

using DotNetCampus.Cli;
using HalnerefaiwhoLanallwaynai;

var options = CommandLine.Parse(args).As<Options>();
```

生成的 CommandLine.As.g.cs 代码如下

```csharp
   file static class Interceptors
   {
       /// <summary>
       /// <see cref="global::DotNetCampus.Cli.CommandLine.As{Options}()"/> 方法的拦截器。拦截以提高性能。
       /// </summary>
       [global::System.Runtime.CompilerServices.InterceptsLocation(1, /* . @Program.cs * / "NiBn1rFX7RQAUKcZwhHx+KYAAABQcm9ncmFtLmNz")]
       public static T CommandLine_As_HalnerefaiwhoLanallwaynaiOptions<T>(this global::DotNetCampus.Cli.CommandLine commandLine)
           where T : global::HalnerefaiwhoLanallwaynai.Options
       {
           return (T)global::HalnerefaiwhoLanallwaynai.OptionsBuilder.CreateInstance(commandLine);
       }
   }
```

最简复现项目： https://github.com/lindexi/lindexi_gd/tree/e6eed4a7d4a8f0aff4f76f0790ff954a89708470/Workbench/HalnerefaiwhoLanallwaynai

--- a/src/DotNetCampus.CommandLine.Analyzer/Generators/BuilderGenerator.cs
+++ b/src/DotNetCampus.CommandLine.Analyzer/Generators/BuilderGenerator.cs
@@ -75,7 +75,7 @@ namespace {{model.Namespace}};
 /// <summary>
 /// 辅助 <see cref="{{model.CommandObjectType.ToGlobalDisplayString()}}"/> 生成命令行选项、谓词或处理函数的创建。
 /// </summary>
-internal sealed class {{model.GetBuilderTypeName()}}
+{{(model.IsPublic ? "public" : "internal")}} sealed class {{model.GetBuilderTypeName()}}
 {
     public static object CreateInstance(global::DotNetCampus.Cli.CommandLine commandLine)
     {

--- a/src/DotNetCampus.CommandLine.Analyzer/Generators/ModelProviding/CommandModelProvider.cs
+++ b/src/DotNetCampus.CommandLine.Analyzer/Generators/ModelProviding/CommandModelProvider.cs
@@ -81,11 +81,13 @@ internal static class CommandModelProvider
 
                 var @namespace = typeSymbol.ContainingNamespace.ToDisplayString();
                 var verbName = attribute?.ConstructorArguments[0].Value?.ToString();
+                var isPublic = typeSymbol.DeclaredAccessibility == Accessibility.Public;
 
                 return new CommandObjectGeneratingModel
                 {
                     Namespace = @namespace,
                     CommandObjectType = typeSymbol,
+                    IsPublic = isPublic,
                     VerbName = verbName,
                     IsHandler = isHandler,
                     OptionProperties = optionProperties,
@@ -131,6 +133,8 @@ internal record CommandObjectGeneratingModel
     public required string Namespace { get; init; }
 
     public required INamedTypeSymbol CommandObjectType { get; init; }
+
+    public required bool IsPublic { get; init; }
 
     public required string? VerbName { get; init; }
 

